src/Features/Core/Portable/DesignerAttribute/DesignerAttributeDiscoveryService.cs

```diff
@@ -156,26 +165,22 @@ private static async ValueTask<bool> HasDesignerCategoryTypeAsync(Project projec
             // change if anything it depends on changes).
             var lazyProjectVersion = AsyncLazy.Create(project.GetSemanticVersionAsync);
 
-            // Switch to frozen semantics if requested.  We don't need to wait on generators to run here as we want to
-            // be lightweight.  We'll also continue running in the future.  So if any changes to happen that are
-            // important to pickup, then we'll see it in the future.  But this avoids constant churn here trying to do
-            // things like building skeletons.
-            if (useFrozenSnapshots)
+            if (specificDocument != null)
             {
-                var document = specificDocument ?? project.Documents.FirstOrDefault();
-                if (document is null)
-                    return;
-
-                project = document.WithFrozenPartialSemantics(cancellationToken).Project;
-                specificDocument = specificDocument is null ? null : project.GetRequiredDocument(specificDocument.Id);
+                // Scan that specific document first.
+                await ScanForDesignerCategoryUsageAsync(
+                    project, specificDocument, callback, lazyProjectVersion, cancellationToken).ConfigureAwait(false);
+
+                // The scan the entire project.  Note: this will no-op on the specific-document as it will have just
+                // computed and cached the data for it.
+                await ScanForDesignerCategoryUsageAsync(
+                    project, specificDocument: null, callback, lazyProjectVersion, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                await ScanForDesignerCategoryUsageAsync(
+                    project, specificDocument: null, callback, lazyProjectVersion, cancellationToken).ConfigureAwait(false);
```

Comment:
@ToddGrun this may be clearer.  but as you can see, both codepaths always do the full scan, which is why i had it originally pulled out.